driver/src/main/com/mongodb/MongoClientOptions.java

```diff
@@ -1233,8 +1240,31 @@ public Builder codecRegistry(final CodecRegistry codecRegistry) {
          * @since 3.1
          */
         public Builder addCommandListener(final CommandListener commandListener) {
-            notNull("commandListener", commandListener);
-            commandListeners.add(commandListener);
+            eventListenerSettingsBuilder.addCommandListener(commandListener);
+            return this;
+        }
+
+        /**
+         * Adds the given connection listener.
+         *
+         * @param connectionListener the non-null connection listener
+         * @return this
+         * @since 3.5
+         */
+        public Builder addConnectionListener(final ConnectionListener connectionListener) {
```

Comment:
I opted to keep the existing methods as in sync and not deprecate them. This seems inline with how settings work in the sync driver.